rclcpp/minimal_action_client/member_function.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <inttypes.h>
+#include <memory>
+#include <string>
+#include <iostream>
+#include "example_interfaces/action/fibonacci.hpp"
+#include "rclcpp/rclcpp.hpp"
+// TODO(jacobperron): Remove this once it is included as part of 'rclcpp.hpp'
+#include "rclcpp_action/rclcpp_action.hpp"
+
+class MinimalActionClient : public rclcpp::Node
+{
+public:
+  using Fibonacci = example_interfaces::action::Fibonacci;
+  using GoalHandleFibonacci = rclcpp_action::ClientGoalHandle<Fibonacci>;
+
+  MinimalActionClient(const std::string node_name)
+    : Node(node_name), goal_done_(false)
+  {
+  }
+
+  void init(const std::string action_name)
+  {
+    this->client_ptr_ =
+      rclcpp_action::create_client<Fibonacci>(this->shared_from_this(), action_name);
```

Comment:
Instead of `shared_from_this()`, you can use the node interfaces for creating a server/client:
https://github.com/ros2/rclcpp/blob/master/rclcpp_action/include/rclcpp_action/create_server.hpp#L56-L68

With this, you can put the logic into the constructor. An example can be found here:
https://github.com/Karsten1987/roscon2018/blob/composition_components/confbot_driver/src/nodes/confbot_driver.cpp#L41-L52

Reply:
Great, thanks for https://github.com/ros2/rclcpp/pull/700 :)